File: AFSwinx/src/main/java/com/tomscz/afswinx/rest/connection/SecurityMethod.java
package com.tomscz.afswinx.rest.connection;

/**
 * This enum holds all supported security method in AFSwinx.
 * @author Martin Tomasek (martin@toms-cz.com)
 *
 * @since 1.0.0.
 */
public enum SecurityMethod {
    BASIC("basic");

    private final String name;

    private SecurityMethod(String name) {
        this.name = name;
    }

    public boolean equalsName(String otherName) {
        return (otherName == null) ? false : name.equals(otherName);
    }

    public String toString() {
        return name;
    }
}

File: AFMobile/AFWinPhone/AFWinPhone/rest/connection/SecurityMethod.cs
using System;

namespace AFWinPhone.rest.connection
{
    public sealed class SecurityMethod
    {

        public static readonly SecurityMethod BASIC = new SecurityMethod("basic");

        private String name;

        private SecurityMethod(String name)
        {
            this.name = name;
        }

        public String toString()
        {
            return name;
        }
    }
}
